src/dataframerow/dataframerow.jl

```diff
@@ -272,6 +272,10 @@ function Base.names(r::DataFrameRow, cols)
     return [string(nms[i]) for i in idxs]
 end
 
+Base.names(r::DataFrameRow, T::Type) =
+    [String(n) for n in _names(r) if eltype(parent(r)[!, n]) <: T]
+Base.names(r::DataFrameRow, fun::Function) = filter(fun, names(r))
```

Comment:
Likewise, why not use `filter!` or `_names` here?

Reply:
Right. The problem is that I used `_names` and it returns a vector of `Symbol` and I have forgotten to add `!`. `names` is guaranteed to freshly allocate a `Vector{String}`.